packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -159,8 +159,11 @@ export class OpGroupingManager {
 			this.config.groupedBatchingEnabled &&
 			// The number of ops in the batch must be 2 or more
 			// or be empty (to allow for empty batches to be grouped)
-			batch.messages.length !== 1
-			// Support for reentrant batches will be on by default
+			// TODO: Can we remove this, as it creates problems for staging mode
+			// as we always want re-submit, even if only 1 op.
+			(batch.messages.length !== 1 ||
+				// Support for reentrant batches will be on by default
+				batch.hasReentrantOps === true)
```

Comment:
```suggestion
			// TODO: Can we remove this, as it creates problems for staging mode
			// as we always want re-submit, even if only 1 op.
			// i'm actually pretty sure there is a bug here.
			// without this change we will not rebase re-entrant ops
			// unless there is more than one op in the batch,
			// but even 1 op must be rebased, as it could be based
			// off a partial state as it was created reentrantly
			// while applying another batch
			(batch.messages.length !== 1 ||
			// Support for reentrant batches will be on by default
			batch.hasReentrantOps === true)
```